ako-gateway-api/k8s/validator.go

```diff
@@ -245,28 +311,39 @@ func isValidListener(key string, gateway *gatewayv1.Gateway, gatewayStatus *gate
 
 	// Valid listener
 	defaultCondition.
-		Reason(string(gatewayv1.GatewayReasonAccepted)).
+		Reason(string(gatewayv1.ListenerReasonAccepted)).
 		Status(metav1.ConditionTrue).
 		Message("Listener is valid").
 		SetIn(&gatewayStatus.Listeners[index].Conditions)
+
+	// Setting the resolvedRef condition
+	resolvedRefCondition.
+		Status(metav1.ConditionTrue).
+		Reason(string(gatewayv1.ListenerReasonResolvedRefs)).
+		Message("All the references are valid").
+		SetIn(&gatewayStatus.Listeners[index].Conditions)
+
 	utils.AviLog.Infof("key: %s, msg: Listener %s/%s is valid", key, gateway.Name, listener.Name)
 	return true
 }
 
-func IsHTTPRouteValid(key string, obj *gatewayv1.HTTPRoute) bool {
+func IsHTTPRouteValid(key string, obj *gatewayv1.HTTPRoute, gateway *gatewayv1.Gateway) bool {
 
 	httpRoute := obj.DeepCopy()
 	if len(httpRoute.Spec.ParentRefs) == 0 {
 		utils.AviLog.Errorf("key: %s, msg: Parent Reference is empty for the HTTPRoute %s", key, httpRoute.Name)
 		return false
 	}
+	httpRouteNsName := httpRoute.Namespace + "/" + httpRoute.Name
+	lib.GetLockSet().Lock(httpRouteNsName)
+	defer lib.GetLockSet().Unlock(httpRouteNsName)
```

Comment:
this lock is not required and can lead to deadlocks.

the locks are required to ensure serial processing of inter dependent gateways and routes, so only locks on gateways should be fine

route status doesn't seem to used for any verification, so as long as the patch calls don't run into race conditions, we should be fine